$NetBSD: patch-configure.ac,v 1.8 2022/11/23 13:10:00 micha Exp $

Patch for Issue 370:
https://gitlab.com/graphviz/graphviz/-/issues/370
https://gitlab.com/graphviz/graphviz/-/issues/2317

Use the same path for all platforms in pkgsrc.

Use lua from pkgsrc.

--- configure.ac.orig	2022-11-19 01:13:27.000000000 +0000
+++ configure.ac
@@ -101,8 +101,15 @@ case "${host_os}" in
 	# For the build number: months since Jan 2000, day of month from the timestamp
 	# For the revision number: hour, minute from the timestamp
 	if test $GRAPHVIZ_VERSION_MICRO != "0"; then
-		GRAPHVIZ_VERSION_BUILD=`expr \( ${GRAPHVIZ_VERSION_MICRO:0:4} - 2000 \) \* 12 + ${GRAPHVIZ_VERSION_MICRO:4:2}`${GRAPHVIZ_VERSION_MICRO:6:2}
-		GRAPHVIZ_VERSION_REVISION=${GRAPHVIZ_VERSION_MICRO:9:4}
+		# JR: patch from Michael van Elst:
+		#     if we have a shell, we should have a working awk, too
+		eval `echo "$GRAPHVIZ_VERSION_MICRO" | awk '{
+			print "GRAPHVIZ_VERSION_BUILD=" \
+				(substr($1,1,4)-2000)*12+substr($1,5,2) \
+				substr($1,7,2)
+			print "GRAPHVIZ_VERSION_REVISION=" \
+				substr($1,10,4)
+		}'`
 	fi
 	;;
 esac
@@ -118,8 +125,7 @@ if test -z "$LIBPOSTFIX"; then
   case "${host_os}" in
     *linux* )
       case "${host_cpu}" in
-        aarch64 | powerpc64 | powerpc64le | s390x | x86_64 | sparc64 | mips64* | riscv64 | e2k )
-          LIBPOSTFIX="64"
+        aarch64* | powerpc64 | powerpc64le | s390x | x86_64 | sparc64 | mips64* | riscv64 | e2k )
           INTGOSIZE=64
           ;;
       esac
@@ -127,7 +133,6 @@ if test -z "$LIBPOSTFIX"; then
     *solaris* )
       case "${host_cpu}" in
         x86_64 | sparc64 )
-          LIBPOSTFIX="/64"
           INTGOSIZE=64
           ;;
       esac
@@ -899,7 +904,7 @@ else
 	            if test "x$?" = "x0" ; then
                         LUA_INCLUDES="$LUA_CFLAGS "`$PKG_CONFIG --cflags lua$l`
                         LUA_LIBS="$LUA_LFLAGS "`$PKG_CONFIG --libs lua$l`
-		        LUA_INSTALL_DIR="/usr/lib$LIBPOSTFIX/lua/`$PKG_CONFIG --variable=V lua`"
+		        LUA_INSTALL_DIR="${PREFIX}/lib$LIBPOSTFIX/lua/`$PKG_CONFIG --variable=V lua`"
 
 		        pkgconfig_lualib_found=`$PKG_CONFIG --exists lualib$l 2>/dev/null`
 		        if test  "x$?" = "x0"; then
@@ -920,7 +925,7 @@ else
 	        echo
                 LUA_INCLUDES=`$LUACONFIG --include`
                 LUA_LIBS=`$LUACONFIG --libs`
-	        LUA_INSTALL_DIR="/usr/lib$LIBPOSTFIX/lua/5.0"
+	        LUA_INSTALL_DIR="${PREFIX}/lib$LIBPOSTFIX/lua/5.0"
             fi
         fi
 
@@ -2096,44 +2101,8 @@ fi
 AM_CONDITIONAL(WITH_QUARTZ, [test "x$use_quartz" = "xYes"])
 
 dnl -----------------------------------
-dnl INCLUDES and LIBS for PLATFORMSDK
-
-AC_ARG_WITH(platformsdkincludedir,
-  [AS_HELP_STRING([--with-platformsdkincludedir=DIR],[use Platform SDK (Windows) includes from DIR])],
-  [PLATFORMSDKINCLUDE=$withval; PLATFORMSDKINCLUDE_ESCAPED=${PLATFORMSDKINCLUDE//'\\'/\\\\}; PLATFORMSDKINCLUDE_ESCAPED=${PLATFORMSDKINCLUDE_ESCAPED// /\\ }])
-
-AC_ARG_WITH(platformsdklibdir,
-  [AS_HELP_STRING([--with-platformsdklibdir=DIR],[use Platform SDK (Windows) libraries from DIR])],
-  [PLATFORMSDKLIB=$withval])
-
-AC_SUBST([PLATFORMSDKINCLUDE])
-AC_SUBST([PLATFORMSDKINCLUDE_ESCAPED])
-AC_SUBST([PLATFORMSDKLIB])
-
-dnl -----------------------------------
 dnl INCLUDES and LIBS for GDI+
 
-AC_ARG_WITH(gdiplus,
-  [AS_HELP_STRING([--with-gdiplus=no],[GDI+ framework (Windows)])],
-  [], [with_gdiplus=no])
-
-if test "x$with_gdiplus" != "xyes"; then
-  use_gdiplus="No (disabled by default - Windows only)"
-else
-  if test -f "$PLATFORMSDKINCLUDE\GdiPlus.h" -a -f "$PLATFORMSDKLIB\GdiPlus.lib"; then
-    use_gdiplus="Yes"
-    AC_DEFINE_UNQUOTED(HAVE_GDIPLUS,1,[Define if you have the GDI+ framework for Windows])
-    GDIPLUS_HEADS=`cd "$PLATFORMSDKINCLUDE" && echo GdiPlus*.h`
-    GDIPLUS_CFLAGS=''
-    GDIPLUS_LIBS=''
-    AC_SUBST([GDIPLUS_HEADS])
-    AC_SUBST([GDIPLUS_CFLAGS])
-    AC_SUBST([GDIPLUS_LIBS])
-  else
-    use_gdiplus="No (missing headers)"
-    AC_MSG_WARN(--with-gdiplus requires valid --with-platformsdkincludedir and --with-platformsdklibdir.)
-  fi
-fi
 AM_CONDITIONAL(WITH_GDIPLUS, [test "x$use_gdiplus" = "xYes"])
 
 dnl -----------------------------------
